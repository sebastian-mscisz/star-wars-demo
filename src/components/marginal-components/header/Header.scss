@import 'src/variables';

.sw-header {
    min-height: $header-minheight;
    background-color: $base-background-color;
    border-right: 2px solid $base-color;
    flex: 0 0 190px;

    @media screen and (max-width: $small-bp) {
        flex-basis: 100%;
        min-height: $header-mobile-minheight;
        height: $header-mobile-height;
        border-right: none;
        border-bottom: 2px solid $base-color;
    }

    &-logo {
        color: black;
        text-shadow:
            -1px -1px 0 #ffee58,
            1px -1px 0 #ffee58,
            -1px 1px 0 #ffee58,
            1px 1px 0 #ffee58;
        text-align: center;
    }
}
