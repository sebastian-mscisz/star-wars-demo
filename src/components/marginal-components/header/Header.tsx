import { NavigationBar } from '../navigation-bar';
import './Header.scss';

export const Header: React.FC = () => {
    return (
        <header className="sw-header">
            <h2 className="sw-header-logo">
                STAR WARS
                <br />
                DEMOPEDIA
            </h2>
            <NavigationBar />
        </header>
    );
};
