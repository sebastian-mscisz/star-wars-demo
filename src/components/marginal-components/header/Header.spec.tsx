import { BrowserRouter } from 'react-router-dom';
import { Header } from './Header';
import { render } from '@testing-library/react';

describe('Header', () => {
    it('should render', () => {
        const { asFragment } = render(<Header />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
