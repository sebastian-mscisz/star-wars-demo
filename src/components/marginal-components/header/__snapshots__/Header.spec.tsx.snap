// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<DocumentFragment>
  <header
    class="sw-header"
  >
    <h2
      class="sw-header-logo"
    >
      STAR WARS
      <br />
      DEMOPEDIA
    </h2>
    <div
      class="sw-nav"
    >
      <ul
        class="sw-nav__links"
      >
        <li
          class="sw-nav__links__item"
        >
          <a
            class="sw-nav__links__item__link"
            href="/characters"
          >
            Characters
          </a>
        </li>
        <li
          class="sw-nav__links__item"
        >
          <a
            class="sw-nav__links__item__link"
            href="/locations"
          >
            Locations
          </a>
        </li>
        <li
          class="sw-nav__links__item"
        >
          <a
            class="sw-nav__links__item__link"
            href="/vehicles"
          >
            Vehicles
          </a>
        </li>
      </ul>
    </div>
  </header>
</DocumentFragment>
`;
